### ISSUE TITLE
Fix upper version numbers in requirements?

### ISSUE THREAD

**Issue by mstimberg** [brian-team/brian2#186]
As we just experienced with sympy 0.7.4, new versions of brian dependencies can break our code. If this happens during a period where we don't make any code changes (i.e. no travis testing is triggered) this can go unnoticed for quite a while.

I wonder whether it'd be better to give a range of versions, e.g. for sympy we would have required a version between 0.7.2 and 0.7.3? We could then use http://requires.io to track outdated dependencies for us, we would test the newest version and then update the dependency. What do you think?

The obvious disadvantage (especially when there's a long time between releases) is that we might annoy users who have a more recent version than the one we tested.

**Comment by thesamovar** [comment]
Yeah, that does seem reasonable. Sigh again. We do have to remember to update that requirement for each release though!

**Comment by mstimberg** [comment]
Another option: do not set a maximum version but raise a warning if the version is newer than the version that we used for testing, something along the lines of "You are using sympy 1.3, but Brian 2.2 has only been tested with versions up to 1.2. If you are experiencing unexpected failures, check whether a new version of Brian 2 has been released or consider downgrading sympy". On the other hand, this could become like Brian 1's scipy warning...

**Comment by thesamovar** [comment]
Potentially we could be a little bit smarter about this. We only use sympy in a few places, perhaps we could catch any exceptions raised from sympy and add a warning at that point only?

**Comment by thesamovar** [comment]
What I mean is: we only warn in the case that an exception is raised, so it wouldn't be as annoying as the Brian 1 scipy warning.
